### PR TITLE
feat(services): add service avatar customization

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1171

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> Don’t Sacrifice Readability, you are going to debug it in 6 months

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(services): add service avatar customization - 2024-08-01T07:51:34Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/ff2ff8d1-1578-4777-898f-8c1db3c3365c/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://1527-console-staging.zc531a994.rustrocks.cloud)

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx [hunk 1]
```diff
@@ -2,7 +2,9 @@ import { BuildModeEnum, type Organization } from 'qovery-typescript-axios'
 import { type FormEventHandler } from 'react'
 import { Controller, useFormContext } from 'react-hook-form'
 import { useNavigate, useParams } from 'react-router-dom'
+import { match } from 'ts-pattern'
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx:5]: useless

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx [hunk 2]
```diff
@@ -2,7 +2,9 @@ import { BuildModeEnum, type Organization } from 'qovery-typescript-axios'
...
 import { AnnotationSetting, LabelSetting } from '@qovery/domains/organizations/feature'
+import { type Application, type Container } from '@qovery/domains/services/data-access'
```
Comment (RemiBonnet) [inline libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx:7]: useless as well

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa, working well!

I don't know if it's a good idea to keep other "standard" icons. For example,  if we have a `DATABASE` it's weird to have the possibility to add `APPLICATION` icon, no? 
Screenshot 2024-08-02 at 11 55 17

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx]: Fixed https://github.com/Qovery/console/pull/1527/commits/935b997887e8833255fc32c2a1ad056e503f213b

File: libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/services/src/lib/ui/page-application-create/step-general/step-general.tsx]: Fixed https://github.com/Qovery/console/pull/1527/commits/935b997887e8833255fc32c2a1ad056e503f213b

File: libs/domains/services/feature/src/lib/service-avatar-switcher/service-avatar-switcher.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+            iconName="angle-down"
+            className="absolute bottom-0 right-0 rounded-full bg-white p-0.5 text-neutral-300"
+          />
+        </button>
+      </Popover.Trigger>
+      <Popover.Content side="bottom" className="grid grid-cols-8 gap-2 text-sm text-neutral-350" style={{ width: 400 }}>
+        {icons.map(({ icon, title, uri }) => {
+          const isSelected = uri === service.icon_uri
+          // XXX: corner case as application and container have the same icon, we want to hide one of them.
+          if (uri === 'app://qovery-console/container') {
```
[the remark this answers is not in the document]
Comment (ctjhoa) [reply libs/domains/services/feature/src/lib/service-avatar-switcher/service-avatar-switcher.tsx:40]: This is a "disguised" enum.
Backend team wanted to have a URI format to anticipate custom icon in the future.
However, this is a special type of URI https://en.wikipedia.org/wiki/List_of_URI_schemes#Unofficial_but_common_URI_schemes where `qovery-console` doesn't represent an internet domain but rather an application name.
Then we use this pattern to have an enum like icon list and we are still in control of available icons.
So the environment as nothing to do with that URI.

**Comment by ctjhoa** [comment]
> Thanks @ctjhoa, working well!
> 
> I don't know if it's a good idea to keep other "standard" icons. For example, if we have a `DATABASE` it's weird to have the possibility to add `APPLICATION` icon, no? Screenshot 2024-08-02 at 11 55 17

I ask Alessandro about it, he says that it could be useful in some rare case and advice to keep it that way.

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**